url587829ef81_champdavis_KhovanovHomology_tgz-pJ8-KhovanovMapJ8/src/SplitMap.java

```diff
@@ -0,0 +1,83 @@
+import java.util.*;
+
+public class SplitMap implements DMap{
+    
+    private KhovanovMapNode fromNode;
+    private KhovanovMapNode toNode;
+    private SeifertCircle fromCircle;
+    private ArrayList<SeifertCircle> toCircles;
+    
+    public SplitMap(KhovanovMapNode f, KhovanovMapNode t, SeifertCircle fc, ArrayList<SeifertCircle> tc){
+        this.fromNode = f;
+        this.toNode = t;
+        this.fromCircle = fc;
+        this.toCircles = tc; 
+        //System.out.println(tc.get(0).getCode());
+        addTensorArrows();
+        System.out.println(toString());
+    }
+    
+    public KhovanovMapNode getToNode(){
+        return this.toNode;
+    }
+    
+    public void addTensorArrows(){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.